extension/source/background/TransactionsController.ts

```diff
@@ -27,78 +24,76 @@ export default class TransactionsController {
     promptUser: async ({ params: [id] }) => {
       const cleanupTasks = new TaskQueue();
 
-      const promptAction = new Promise<string | undefined>(
-        (resolve, _reject) => {
-          (async () => {
-            const lastWin = await windows.getLastFocused();
-
-            const popupWidth = 400;
-            let left: number | undefined;
-
-            if (lastWin.width !== undefined && lastWin.left !== undefined) {
-              left = lastWin.left + lastWin.width - popupWidth - 20;
+      const promptAction = new Promise<
+        TransactionStatus.APPROVED | TransactionStatus.REJECTED
+      >((resolve, _reject) => {
+        (async () => {
+          const lastWin = await windows.getLastFocused();
+
+          const popupWidth = 400;
+          let left: number | undefined;
+
+          if (lastWin.width !== undefined && lastWin.left !== undefined) {
+            left = lastWin.left + lastWin.width - popupWidth - 20;
+          }
+
+          const popup = await windows.create({
+            url: runtime.getURL(`confirm.html?id=${id}`),
+            type: 'popup',
+            width: popupWidth,
+            height: 700,
+            left,
+          });
+
+          cleanupTasks.push(() => {
+            if (popup.id !== undefined) {
+              windows.remove(popup.id);
             }
+          });
 
-            const popup = await windows.create({
-              url: runtime.getURL(`confirm.html?id=${id}`),
-              type: 'popup',
-              width: popupWidth,
-              height: 700,
-              left,
-            });
-
-            cleanupTasks.push(() => {
-              if (popup.id !== undefined) {
-                windows.remove(popup.id);
-              }
-            });
-
-            function onRemovedListener(windowId: number) {
-              if (windowId === popup.id) {
-                resolve(undefined);
-              }
+          function onRemovedListener(windowId: number) {
+            if (windowId === popup.id) {
+              resolve(TransactionStatus.REJECTED);
```

Comment:
Closing the dialog implicitly rejects the transaction. We could add something more granular but `undefined` doesn't seem right.